packages/dd-trace/src/config.js

```diff
@@ -806,31 +544,289 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
     tagger.add(tags, DD_TRACE_TAGS)
     tagger.add(tags, DD_TRACE_GLOBAL_TAGS)
 
-    this._setString(env, 'service', DD_SERVICE || DD_SERVICE_NAME || tags.service)
+    this._setValue(env, 'appsec.blockedTemplateHtml', maybeFile(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML))
+    this._setValue(env, 'appsec.blockedTemplateJson', maybeFile(DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON))
+    this._setBoolean(env, 'appsec.enabled', DD_APPSEC_ENABLED)
+    this._setString(env, 'appsec.obfuscatorKeyRegex', DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP)
+    this._setString(env, 'appsec.obfuscatorValueRegex', DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP)
+    this._setValue(env, 'appsec.rateLimit', maybeInt(DD_APPSEC_TRACE_RATE_LIMIT))
+    this._setString(env, 'appsec.rules', DD_APPSEC_RULES)
+    this._setBoolean(env, 'appsec.customRulesProvided', !!DD_APPSEC_RULES)
```

Comment:
otherwise customRulesProvided will always be set from env, so default won't be ever used
```suggestion
    if (DD_APPSEC_RULES) this._setBoolean(env, 'appsec.customRulesProvided', !!DD_APPSEC_RULES)
```

Reply:
Also it's more consistent with the programming config one in line 638

Reply:
I think it's fine to never use the default, since ['DD_APPSEC_RULES'](this._setBoolean(env, 'appsec.customRulesProvided', !!DD_APPSEC_RULES)) doesn't have a default. I set a default for it because [`_merge`](https://github.com/DataDog/dd-trace-js/blob/0fcd44e0935c904d290783c902c4d49f8ed1df24/packages/dd-trace/src/config.js#L914) uses `this._defaults` as keys

Reply:
You made this change yourself in prog config here: https://github.com/DataDog/dd-trace-js/pull/3928/commits/a0df2a2e45cf93f367d6460630835fddb9b77160
why not in the env var config too ?

Reply:
because env var is the last parameter in coalesce, which it falls back to if it receives no valid value. so we can ignore default

Reply:
I don't understand what you mean, let's figure this out over zoom

Reply:
From our conversation: put back the `if (config.appsec.rules) setBool....` because otherwise the absence of env var will always override the default